src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -70,9 +70,17 @@ def trace(self) -> JinjaTrace:
         trace_template = self.make_template(trace_template_str)
         trace_template_output = trace_template.render()
         # Split output by section. Each section has two possible formats.
-        for p in trace_template_output.split("\0"):
```

Comment:
Previously, each output entry was terminated by a `\0` character. In order to handle the new case, it proved better to **prefix** each output entry with a `\0` instead, hence I refactored the loop.